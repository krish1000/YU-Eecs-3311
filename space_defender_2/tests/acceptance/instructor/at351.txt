--Test destroying enemy interceptor. Also tests enemy blocking spawn.
toggle_debug_mode
play(5,10,1,1,1,38,38)

setup_next(5)
move(A,1)
move(A,2)
fire
fire
fire
fire
