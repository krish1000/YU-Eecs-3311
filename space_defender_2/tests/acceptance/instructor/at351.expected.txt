  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->toggle_debug_mode
  state:not started, debug, ok
  In debug mode.
->play(5,10,1,1,1,38,38)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_next(5)
  state:in game(0.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:70/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[C,1]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(A,1)
  state:in game(1.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:64/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[A,1]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,10]
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [C,1] -> [A,1]
  Enemy Action:
  Natural Enemy Spawn:
    A Interceptor(id:1) spawns at location [A,10].
      1  2  3  4  5  6  7  8  9 10
    A S  _  _  _  _  _  _  _  _  I
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(A,2)
  state:in game(2.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:64/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[A,2]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,7]
    [2,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,10]
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [A,1] -> [A,2]
  Enemy Action:
    A Interceptor(id:1) moves: [A,10] -> [A,7]
  Natural Enemy Spawn:
    A Interceptor(id:2) spawns at location [A,10].
      1  2  3  4  5  6  7  8  9 10
    A _  S  _  _  _  _  I  _  _  I
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(3.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:62/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[A,2]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,7]
    [2,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,10]
  Projectile:
    [-1,*]->damage:70, move:5, location:[A,3]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [A,2].
      A friendly projectile(id:-1) spawns at location [A,3].
  Enemy Action:
    A Interceptor(id:1) stays at: [A,7]
    A Interceptor(id:2) stays at: [A,10]
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  S  *  _  _  _  I  _  _  I
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(4.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:60/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[A,2]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:1
  Enemy:
    [2,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,10]
    [3,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[B,10]
  Projectile:
    [-2,*]->damage:70, move:5, location:[A,3]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [A,3] -> [A,7]
      The projectile collides with Interceptor(id:1) at location [A,7], dealing 70 damage.
      The Interceptor at location [A,7] has been destroyed.
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [A,2].
      A friendly projectile(id:-2) spawns at location [A,3].
  Enemy Action:
    A Interceptor(id:2) stays at: [A,10]
  Natural Enemy Spawn:
    A Interceptor(id:3) spawns at location [B,10].
      1  2  3  4  5  6  7  8  9 10
    A _  S  *  _  _  _  _  _  _  I
    B _  _  _  _  _  _  _  _  _  I
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(5.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:58/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[A,2]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:1
  Enemy:
    [2,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,10]
    [3,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[B,10]
  Projectile:
    [-2,*]->damage:70, move:5, location:[A,8]
    [-3,*]->damage:70, move:5, location:[A,3]
  Friendly Projectile Action:
    A friendly projectile(id:-2) moves: [A,3] -> [A,8]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [A,2].
      A friendly projectile(id:-3) spawns at location [A,3].
  Enemy Action:
    A Interceptor(id:2) stays at: [A,10]
    A Interceptor(id:3) stays at: [B,10]
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  S  *  _  _  _  _  *  _  I
    B _  _  _  _  _  _  _  _  _  I
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(6.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:56/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[A,2]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:2
  Enemy:
    [3,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,10]
  Projectile:
    [-3,*]->damage:70, move:5, location:[A,8]
    [-4,*]->damage:70, move:5, location:[A,3]
  Friendly Projectile Action:
    A friendly projectile(id:-2) moves: [A,8] -> [A,10]
      The projectile collides with Interceptor(id:2) at location [A,10], dealing 70 damage.
      The Interceptor at location [A,10] has been destroyed.
    A friendly projectile(id:-3) moves: [A,3] -> [A,8]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [A,2].
      A friendly projectile(id:-4) spawns at location [A,3].
  Enemy Action:
    A Interceptor(id:3) moves: [B,10] -> [A,10]
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  S  *  _  _  _  _  *  _  I
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
